File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());

        // int i = 0;
        // int j = 1;
        // int k = 0;
        // while (k < n / 2 && i < n) {
        //     if (nums[i] == nums[j] && j < n) {
        //         k++;
        //         j++;
        //     }
        //     else {
        //         k = 0;
        //         i++;
        //     }
        // }
        // return nums[i];
        return nums[n/2];
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length-1;
        int j = 0;
        while (j <= n/2){
            int counter = 0;
            for(int i = 0; i <= n; i++){
                if(nums[j] == nums[i] ){
                    counter++;
                }
            }
            if(counter > n/2){
                return nums[j];
            }
            j++;
        }
        return nums[0];
    }
}